### PR TITLE
split of MDNSAnnouncer in order to remove optional dependency

### LINKED ISSUE #2066: io.rest sometimes throws a NCDFE
io.rest declares its dependency to io.transport.mdns as optional. If the dependency is not resolved, this can result in NCDFEs:
```
16:21:39.652 [ERROR] [org.eclipse.smarthome.io.rest       ] - FrameworkEvent ERROR - org.eclipse.smarthome.io.rest
java.lang.NoClassDefFoundError: org/eclipse/smarthome/io/transport/mdns/MDNSService
	at java.lang.Class.getDeclaredMethods0(Native Method)[:1.8.0_40]
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)[:1.8.0_40]
	at java.lang.Class.getDeclaredMethods(Class.java:1975)[:1.8.0_40]
	at org.apache.felix.scr.impl.helper.BindMethod.getServiceObjectAssignableMethod(BindMethod.java:454)
```

### PR DESCRIPTION
**Pull request by kaikreuzer**
fixes #2066

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest.mdns/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Eclipse SmartHome REST MDNS Announcer
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest.mdns/META-INF/MANIFEST.MF]: mDNS?

File: bundles/io/org.eclipse.smarthome.io.rest.mdns/pom.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <artifactId>io</artifactId>
+    <version>0.9.0-SNAPSHOT</version>
+  </parent>
+
+
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.eclipse.smarthome.io</groupId>
+  <artifactId>org.eclipse.smarthome.io.rest.mdns</artifactId>
+
+  <name>Eclipse SmartHome REST MDNS Announcer</name>
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest.mdns/pom.xml]: mDNS?

**Comment by maggu2810** [comment]
Just a minor comment to change MDNS to mDNS in human readable descriptions.

**Comment by kaikreuzer** [comment]
ok, fixed!

**Pull request merged**